### ISSUE TITLE
[Planned feature] Implement partial path with final trade mode

### ISSUE THREAD

**Issue by Reidmcc** [Reidmcc/rockfish#4]
**Describe the solution you'd like**

I want to have a mode where the path payment is two of the three trades and leave the last step as an actual offer. This would allow the bot to take advantage of the spread on the last step; taker-taker-maker instead of taker-taker-taker. It may also make the path payments more reliable since they will have fewer steps.

Requirements: 
- Generalize the path-payment functions to create *n*-step paths instead of hard coded three-step paths
- Add ability to place and track the final-step trades
- Implement fill tracking
- This mode should be able to manage trades for *n* assets

**Comment by ilyacherevkov** [comment]
Better to make passive arbitrage on the first step.

You constantly move your offer at the edge of the spread while maintaining final target margin given you execute last two steps at market at current bid/ask.

This is a less riskier approach, which greatly reduces the risk to end up in some shitty coin with no chance of recovery which I did a lot with all-steps-active arbitrage. 

Since stellar has tiny commissions which have almost no impact on margin, it can actually work, at least it works on waves dex to generate some tiny income.

You can also whitelist assets with great liquidity to almost completely eliminate risk, but it might impact frequency of trades.

**Comment by Reidmcc** [comment]
@ilyacherevkov good point on making the first step the passive step. As for white-listing, Rockfish currently requires you to specify assets, and I plan to keep that in place for any future modes.
